sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```
@@ -704,16 +814,91 @@ public HttpPipeline getHttpPipeline() {
+        return Flux.fromIterable(queueMessageInternalItems)
+            .flatMap(queueMessageItemInternal ->
+                transformQueueMessageItemInternal(queueMessageItemInternal, messageEncoding)
+                .onErrorResume(IllegalArgumentException.class, e -> {
+                    if (processMessageDecodingErrorAsyncHandler != null) {
+                        return transformQueueMessageItemInternal(
+                            queueMessageItemInternal, QueueMessageEncoding.NONE)
+                            .flatMap(messageItem -> processMessageDecodingErrorAsyncHandler.apply(
+                                new QueueMessageDecodingError(
+                                    this, new QueueClient(this),
+                                    messageItem, null, e)))
+                            .then(Mono.empty());
+                    } else if (processMessageDecodingErrorHandler != null) {
+                        return transformQueueMessageItemInternal(
+                            queueMessageItemInternal, QueueMessageEncoding.NONE)
+                            .flatMap(messageItem -> {
+                                try {
+                                    processMessageDecodingErrorHandler.accept(
+                                        new QueueMessageDecodingError(
+                                            this, new QueueClient(this),
+                                            messageItem, null, e));
+                                    return Mono.<QueueMessageItem>empty();
+                                } catch (RuntimeException re) {
+                                    return FluxUtil.<QueueMessageItem>monoError(logger, re);
+                                }
+                            })
+                            .subscribeOn(Schedulers.boundedElastic());
+                    } else {
+                        return FluxUtil.monoError(logger, e);
+                    }
+                }))
+            .collectList()
```

At this point, when a message fails to be transformed do we want to return an error or omit the failed message and continue attempting to transform the remaining messages?

---

If handler isn't provided we want it to fail and let user know they have a problem.